hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/ha/RequestHedgingProxyProvider.java

```diff
@@ -140,14 +150,36 @@ public Object call() throws Exception {
         }
       }
     }
-  }
 
+    /**
+     * Get the successful/Left proxy
+     * If a proxy has failed, we, either, return the successful one, or the one not yet tested.
```

Comment:
This comment sounds confusing -> "if the proxy has failed we return the successful one" that does not fit with the fact that this method is called at the very beginning even before trying any proxy at all

Reply:
Done!